### PR TITLE
Offline tooling for training to use reduction with keepdims=False

### PR DESCRIPTION
**Pull request by baijumeswani**
Fixes https://github.com/microsoft/onnxruntime/issues/19008

The offline tooling currently outputs the loss after reduction with the dimensions intact. Which means a tensor with shape [x, y, z] would result in a loss of [1, 1, 1] and not a scalar output.

This pull request makes the output with `keepdims=False`.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_ort_apis_onnxblock.py [hunk 1]
```diff
@@ -1017,3 +1017,29 @@ def test_save_ort_format():
...
+            self._add = onnxblock.blocks.Add()
+
+        def build(self, input1, input2):
+            return self._add(self._loss1(input1, target_name="target1"), self._loss2(input2, target_name="target2"))
+
+    model = ModelWithTwoOutputs()
+    onnx_model = _get_onnx_model(model, (torch.randn(20, 100, 35, 45), torch.randn(40, 100, 70)))
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        artifacts.generate_artifacts(onnx_model, loss=CustomLossBlock(), artifact_directory=temp_dir)
```
Comment (askhade) [inline orttraining/orttraining/test/python/orttraining_test_ort_apis_onnxblock.py:1045]: What is the success criteria for this test? If the artifacts are generated and no error is thrown, is that considered as a successful test?

Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_ort_apis_onnxblock.py:1045]: No error raised is the success criteria. Any error raised while calling `artifacts.generate_artifacts` will result in pytest test failure.

**Review by askhade — APPROVED** [review]
Overall LGTM. Will be good to add a comment in the test explaining what this new test is testing.

**Review by askhade — DISMISSED** [review]

**Review by askhade — APPROVED** [review]

**Comment by baijumeswani** [comment]
Thank you for the review! :)

**Pull request merged**